test/test_helpers.go

```diff
@@ -0,0 +1,82 @@
+package test
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/files"
+	"github.com/gruntwork-io/terratest/modules/random"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"os"
+	"os/user"
+	"strings"
+)
+
+// runID() Creates a unique ID suitable for including in the name of any cloud resource such that tests can be
+// parallelized.  However, if there is an ENV set with the prefix "SKIP_", then assume we are working on individual tests
+// locally and use the current username as the id.
+func runID() (string, error) {
+	if test_structure.SkipStageEnvVarSet() {
+		currentUser, err := user.Current()
+		if err != nil {
+			return "", err
+		}
+		return strings.ToLower(currentUser.Username), nil
+	}
+
+	return strings.ToLower(random.UniqueId()), nil
+}
+
+// copySupportingFiles copies one or more files from the test/ dir into a destination dir.
+// This is done to configure providers when using modules without them explicitly defined.
+func copySupportingFiles(fileNames []string, destination string) error {
+	testFileSourceDir, getTestDirErr := os.Getwd()
+	if getTestDirErr != nil {
+		fmt.Println("Calling t.FailNow(): could not execute os.Getwd(): ", getTestDirErr)
+		return getTestDirErr
+	}
+
+	fmt.Println("Test working directory is: ", testFileSourceDir)
+
+	fmt.Println("Copying files: ", fileNames, " to temporary test dir: ", destination)
+	for _, file := range fileNames {
+		src := testFileSourceDir + "/" + file
+		dest := destination + "/" + file
+		copyErr := files.CopyFile(src, dest)
+		if copyErr != nil {
+			fmt.Println("😩 Calling t.FailNow(): failed copying from: ", src, " to: ", dest, " with error: ", copyErr)
+			return copyErr
+		} else {
+			fmt.Println("✌️ Success! Copied from: ", src, " to: ", dest)
+		}
+	}
+	return nil
+}
+
+// cleanupSupportingFiles deletes one or more files from a directory, intended to be called after copySupportingFiles
+func cleanupSupportingFiles(fileNames []string, destination string) error {
+	fmt.Println("Deleting files: ", fileNames, "from dir: ", destination)
+	for _, file := range fileNames {
+		fullPath := destination + "/" + file
+		removeErr := os.Remove(fullPath)
+		if removeErr != nil {
+			fmt.Println("😩 Failed deleting file ", fullPath, " with error: ", removeErr)
+			return removeErr
+		} else {
+			fmt.Println("✌️ Success! Deleted file: ", fullPath)
+		}
+	}
+	return nil
+}
+
+// getGoogleCredentials reads a static service account credentials JSON file named gcp-creds.json from the test/ folder
+func getGoogleCredentials() string {
+	envGoogleCredentials, envPresent := os.LookupEnv("GOOGLE_CREDENTIALS")
```

Comment:
I feel like I should point out the existence of terratest's [GetGoogleCredentialsFromEnvVar](https://pkg.go.dev/github.com/gruntwork-io/terratest/modules/gcp#GetGoogleCredentialsFromEnvVar)

Reply:
Once there is some time to breath post launch I am making a proper template repo that has tests with consistent patterns for all this mapped out then backtrack through existing repos and make them all follow it.  Implementation of tests is tedious because of super inconsistent test to reference from and awful docs on Terratest's side.

Also their code is kinda crap at times, eg on [that function](https://github.com/gruntwork-io/terratest/blob/v0.40.3/modules/gcp/provider.go#L8-L14) they have copied [a slice](https://github.com/hashicorp/terraform-provider-google/blob/master/google/provider_test.go#L39-L45) from the google terraform provider without realizing that `GOOGLE_USE_DEFAULT_CREDENTIALS` is used as a bool flag and should never contain credentials.